include/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/IERS/Manager.hpp

```diff
@@ -95,52 +95,38 @@ class Manager
 
     Directory getFinals2000ADirectory() const;
 
-    /// @brief              Get array of Bulletin A
+    /// @brief              Get Bulletin A
     ///
-    /// @return             Array of Bulletin A
-
-    Array<BulletinA> getBulletinAArray() const;
-
-    /// @brief              Get Bulletin A at instant
-    ///
-    /// @param              [in] anInstant An instant
     /// @return             Bulletin A
 
-    BulletinA getBulletinAAt(const Instant& anInstant) const;
-
-    /// @brief              Get array of Finals 2000A
-    ///
-    /// @return             Array of Finals 2000A
-
-    Array<Finals2000A> getFinals2000AArray() const;
+    BulletinA getBulletinA();
 
-    /// @brief              Get Finals 2000A at instant
+    /// @brief              Get Finals 2000A
     ///
-    /// @param              [in] anInstant An instant
     /// @return             Finals 2000A
 
-    Finals2000A getFinals2000AAt(const Instant& anInstant) const;
+    Finals2000A getFinals2000A();
```

Comment:
```suggestion
    Finals2000A getFinals2000A() const;
```
I believe it's worth marking these as const, as the relevant member variables that changed are marked as mutable.

Reply:
But they're not marked as mutable anymore, it seems Kyle removed those variables? And having the "intention" shown at the interface level is better than lying at the interface and then allowing mutation of the private vars right?

Reply:
This is a really interesting thing to bring up. Thanks for calling it out @vishwa2710. I looked into why and when you should use `mutable` with `const` and it has a lot to do with concurrency in C++.

Really good article about it written by the chair of the C++ standards committee:
https://herbsutter.com/2013/05/24/gotw-6a-const-correctness-part-1-3/

It mostly boils down to:

> Using const consistently is simply necessary for correctly-synchronized code.

> If your class contains a member that could change even for const objects and operations, make that member mutable and protect it with a mutex or make it atomic.

Having `const` member functions is important for parallel access to the `Manager` object. It allows multiple threads to know that they can access the object concurrently. Writes to an object by `const` member functions is okay but only if the member is mutable and protected by a mutex (called the M&M rule).

So at the end of the day, I do think we want these member functions to be marked `const`, but we **do not** want to `const_cast` everywhere like we were doing in some places. This should be handled in the proper way by marking things as mutable.

TIL 🌈

Reply:
Yup okay agreed